sycl/plugins/unified_runtime/ur/adapters/level_zero/ur_level_zero.hpp

```diff
@@ -107,8 +107,6 @@ static ur_result_t ze2urResult(ze_result_t ZeResult) {
       {ZE_RESULT_ERROR_OVERLAPPING_REGIONS, UR_RESULT_ERROR_INVALID_OPERATION},
       {ZE_RESULT_ERROR_INVALID_GROUP_SIZE_DIMENSION,
        UR_RESULT_ERROR_INVALID_WORK_GROUP_SIZE},
-      {ZE_RESULT_ERROR_MODULE_BUILD_FAILURE,

```

Comment:
why did you remove this?

Reply:
`UR_RESULT_ERROR_MODULE_BUILD_FAILURE`  was removed from UR spec.  I do not know why and I don't see an obvious replacement

https://github.com/oneapi-src/unified-runtime/commit/d7e4d2b67c56123b37fe7b3d558a70ba853cd311

Reply:
@smaslov-intel : change from @bmyates  is correct. Once this merged, I will rebased https://github.com/intel/llvm/pull/8402 with latest changes in interfaces.

Reply:
@jandres742 : but you will need this error in UR for here: https://github.com/intel/llvm/blob/30c80625ad339aadbb49156c2cc3eef140d7b0dd/sycl/plugins/level_zero/pi_level_zero.cpp#LL3713C2-L3713C2

Reply:
@smaslov-intel : ah I see what you mean. You mean we cannot remove the entry but replace it. See below to what I actually did while testing @bmyates  patch locally:

```cpp
     {ZE_RESULT_ERROR_MODULE_BUILD_FAILURE,
-     UR_RESULT_ERROR_MODULE_BUILD_FAILURE},
+     UR_RESULT_ERROR_PROGRAM_BUILD_FAILURE},
```

is that what you mean?

Reply:
> @smaslov-intel : ah I see what you mean. You mean we cannot remove the entry but replace it. See below to what I actually did while testing @bmyates patch locally:
> 
> ```c++
>      {ZE_RESULT_ERROR_MODULE_BUILD_FAILURE,
> -     UR_RESULT_ERROR_MODULE_BUILD_FAILURE},
> +     UR_RESULT_ERROR_PROGRAM_BUILD_FAILURE},
> ```
> 
> is that what you mean?

Yep, looks like they just renamed module to program, so we need to replace.